demo_proj_credilio/app/controllers/Http/UserController.ts

```diff
@@ -0,0 +1,217 @@
+        .first();
+      if (profileExists)
+        return { message: "Profile already exists", data: profileExists };
+
+      const profile = await Profile.create(payload);
+
+      return { message: "Profile created successfully", data: profile };
+    } catch (e) {
+      return { message: e.messages };
+    }
+  }
+
+  public async viewProfile({
+    auth,
+  }: HttpContextContract): Promise<profileResponseType> {
+    try {
+      const userId = auth.user?.id;
+      const profile = await Profile.query()
+        .where({ user_id: userId })
+        .select("name", "userId", "gender", "dob")
+        .first();
+
+      if (!profile) return { message: "profile is not created yet." };
+
+      await profile.load("user", (userQuery) => {
+        userQuery.select("email");
+      });
+
+      let resObj = {
+        message: "profile fetched successfully.",
+        data: {
+          name: profile.name,
+          email: profile.user.email,
+          gender: profile.gender,
+          dob: profile.dob,
+        },
+      };
+
+      return resObj;
+    } catch (e) {
+      return { message: e.messages };
+    }
+  }
+  public async updateProfile({
+    auth,
+    request,
+  }: HttpContextContract): Promise<profileResponseType> {
+    const userId = auth.user?.id;
+
+    const newProfileSchema = schema.create({
+      name: schema.string.optional([rules.minLength(3), rules.maxLength(30)]),
+      mobile: schema.string.optional([rules.mobile()]),
+      gender: schema.string.optional([rules.regex(new RegExp(/male|female/i))]),
+      dob: schema.date.optional({ format: "yyyy-MM-dd" }),
+    });
+    try {
+      const payload = await request.validate({ schema: newProfileSchema });
+
+      if (!payload || Object.keys(payload).length === 0)
+        throw new Error("require at least one valid field");
+      else {
+        const profile = await Profile.findBy("user_id", userId);
+        if (!profile) throw new Error("userId not found");
+
+        const updatedProfile = await profile?.merge(payload).save();
+
+        let resObj = {
+          message: "profile updated",
+          data: {
+            name: updatedProfile?.name,
+            mobile: updatedProfile?.mobile,
+            gender: updatedProfile?.gender,
+            dob: updatedProfile?.dob,
+          },
+        };
+
+        return resObj;
+      }
+    } catch (e) {
+      return { message: e.messages ? e.messages : e.message };
+    }
+  }
+
+  public async deleteProfile({
+    auth,
+  }: HttpContextContract): Promise<profileResponseType> {
+    try {
+      const userId = auth.user?.$original.id;
+
+      const isProfileDeleted = await Profile.query()
+        .where("user_id", userId)
+        .delete();
+      if (!isProfileDeleted[0]) throw new Error("No profile exists");
+
+      const isUserDeleted = await User.query().where("id", userId).delete();
+      if (isUserDeleted[0]) return { message: "user deleted" };
+      else return { message: "user not found" };
```

Comment:
requirement is to delete the profile only. Not the user.
Its better to first find the profile, if found delete it else throw error that profile not found.

Reply:
it is said in the doc to remove the user as well

Reply:
done